### PR TITLE
Adds approvals before merge and merge if discussions resolved fields to Projects

### PR DESCRIPTION
**Pull request by paulocezar**
Adds `only_allow_merge_if_all_discussions_are_resolved` and `approvals_before_merge` fields to Project and EditProjectOptions structs

### CODE REVIEW

File: projects.go [hunk 1]
```diff
@@ -34,43 +34,45 @@ type ProjectsService struct {
...
+	TagList                                   []string             `json:"tag_list"`
+	Owner                                     *User                `json:"owner"`
+	Name                                      string               `json:"name"`
+	NameWithNamespace                         string               `json:"name_with_namespace"`
+	Path                                      string               `json:"path"`
+	PathWithNamespace                         string               `json:"path_with_namespace"`
+	IssuesEnabled                             bool                 `json:"issues_enabled"`
+	OpenIssuesCount                           int                  `json:"open_issues_count"`
+	MergeRequestsEnabled                      bool                 `json:"merge_requests_enabled"`
+	ApprovalsBeforeMerge                      int                  `json:"approvals_before_merge"`
```
Comment (svanharmelen) [inline projects.go]: I cannot seem to find this field (`ApprovalsBeforeMerge`) in the API docs. Could you point me to right direction so I can confirm this one?

Comment (paulocezar) [inline projects.go]: Yeah, it's actually not documented on the Projects API. It's description is on the [Merge Requests API](https://docs.gitlab.com/ee/api/merge_requests.html#create-mr) and there's a mention there of its presence on the project.

Comment (svanharmelen) [inline projects.go]: Check... Thanks...

**Comment by svanharmelen** [comment]
@paulocezar thanks for the PR!

**Pull request merged**